docarray/array/mixins/find.py

```diff
@@ -125,11 +126,15 @@ def find(
                       parameter is ignored. By default, the Document `text` attribute will be used for search,
                       otherwise the tag field specified by `index` will be used. You can only use this parameter if the
                       storage backend supports searching by text.
+        :param on: specifies a subindex to search on. If set, then the returned DocumentArray will be retrieved from the given subindex. Needs to be supported by a given Document Store.
         :param kwargs: other kwargs.
 
         :return: a list of DocumentArrays containing the closest Document objects for each of the queries in `query`.
         """
 
+        if on is not None:
+            kwargs.update({'on': on})
```

Comment:
I prefer to pass `on=on` wherever is needed, or just avoid having `on` as a keyword parameter here